sdk/monitor/monitor-opentelemetry/src/index.ts

```diff
@@ -57,19 +60,51 @@ export function useAzureMonitor(options?: AzureMonitorOpenTelemetryOptions) {
   // Initialize OpenTelemetry SDK
   const sdkConfig: Partial<NodeSDKConfiguration> = {
     autoDetectResources: true,
-    logRecordProcessor: logHandler.getLogRecordProcessor(),
     metricReader: metricHandler.getMetricReader(),
     views: metricHandler.getViews(),
     instrumentations: instrumentations,
     resource: config.resource,
     sampler: traceHandler.getSampler(),
-    spanProcessor: traceHandler.getSpanProcessor(),
   };
   sdk = new NodeSDK(sdkConfig);
   sdk.start();
+
+  // TODO: Send processors as NodeSDK config once arrays are supported
+  // https://github.com/open-telemetry/opentelemetry-js/issues/4451
+
+  // Add extra SpanProcessors, MetricReaders and LogRecordProcessors
+  let spanProcessors: SpanProcessor[] = options?.spanProcessors || [];
+  spanProcessors.push(traceHandler.getAzureMonitorSpanProcessor());
+  // Add batch processor as the last one
+  spanProcessors.push(traceHandler.getBatchSpanProcessor());
+
   // Add extra SpanProcessors, MetricReaders and LogRecordProcessors
```

Comment:
```suggestion

```

Reply:
If I am understanding this correctly, this always appends the azure processor to the end. In other words:

- if user's processor list already includes azure, it will be double exported
- Users cannot choose to disable the azure processor
- Users cannot choose the specific placement of the azure processor

It'd be best if Python and Node agreed on this and defaulting behavior. I'd prefer having a default value of our azure processor and then letting customers completely override it (including removing the processor) if they want. Cleaner and more customizable.

Reply:
@lzchen What do you think?

Reply:
We don't expose our processor so customers should never create an instance of it, if customer add another processor exporting somewhere else it would be grabbed even if is sending to Breeze as well, Azure Distro without our own processors is just a generic OTel distro with no Azure specific stuff, what is the scenario of customers wanting to remove our processors, should they use another distro instead?

Reply:
I think the experience for customers should be "pass in a list of custom processors that i want to run before batchspanprocessor" and then append batchspanprocessor automatically in distro as @hectorhdzg  already has. This feature for defining a list of custom spanprocessors should only be for custom processors and for modifying telemetry use case. 

>  letting customers completely override it (including removing the processor) if they want.

There shouldn't be a case in which customers want to remove our processor. If they want to use a different exporting mechanism, they shouldnt be using the distro.